### PR TITLE
API root URL is build specific

### PR DESCRIPTION
**Pull request by tanrenyi**
Trello card: https://trello.com/c/chVd3GBa

**Changes:**
- Reconfigure webpack to have different env config
- API URL is build specific

On testing env, url: `https://bucketlist-testing.herokuapp.com/api/v1`
On staging env, url: `https://bucketlist-mvp-staging.herokuapp.com/api/v1`
On production env, url: `https://bucketlist-mvp.herokuapp.com/api/v1`
On local evn, url: `http://localhost:3001/api/v1`

**Using `.env` for SECRETS**
- remove `config/loginSecrets.js` and `config/mysql_user.json`
- all secrets are stored in `.env` file which members have to download and put into the root dir
- secrets are also stored in heroku config vars which will inject into the app during build heroku's phase and be exposed via `process.env`

### CODE REVIEW

File: build-utils/webpack.common.js [hunk 1]
```diff
@@ -1,13 +1,46 @@
...
 const CopyWebpackPlugin = require('copy-webpack-plugin');
-var ServiceWorkerWebpackPlugin = require('serviceworker-webpack-plugin');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+const ServiceWorkerWebpackPlugin = require('serviceworker-webpack-plugin');
+
+
+const SECRETS = [
+  "MYSQL_PASSWORD",
+  "GOOGLE_LOGIN_SECRET",
+  "FACEBOOK_LOGIN_SECRET",
```
Comment (karenfrilya97) [inline build-utils/webpack.common.js]: Can we rename this to `FACEBOOK_APP_ID`?

File: build-utils/webpack.common.js [hunk 2]
```diff
@@ -1,13 +1,46 @@
...
+  "FACEBOOK_APP"
```
Comment (karenfrilya97) [inline build-utils/webpack.common.js]: Can we rename this to `FACEBOOK_APP_SECRET`?

File: .gitignore [hunk 3]
```diff
@@ -2,3 +2,4 @@
 /build
 /config
```
[the remark this answers is not in the document]
Comment (tanrenyi) [reply .gitignore]: yes. one thing, i want to make sure we're not using pem files in the codebase. can you help me clarify?

File: src/client/constants/apis.js [hunk 4]
```diff
@@ -1,5 +1,12 @@
-//const root = 'http://localhost:3001/api/v1'
-const root = 'https://bucketlist-pwa.herokuapp.com/api/v1'
+let root;
+const baseURL = process.env.APP_NAME;
+
+if (process.env.IS_HEROKU === 'true') {
+  root = `https://${baseURL}.herokuapp.com/api/v1`;
```
[the remark this answers is not in the document]
Comment (tanrenyi) [reply src/client/constants/apis.js]: resolved

File: build-utils/webpack.dev.js [hunk 5]
```diff
@@ -7,7 +7,6 @@ module.exports = {
     new DefinePlugin({
       'process.env': {
         'NODE_ENV': JSON.stringify('development'),
```
Comment (karenfrilya97) [inline build-utils/webpack.dev.js]: This extra comma will cause an error right?

File: src/server/config/settings.js [hunk 6]
```diff
@@ -15,10 +15,5 @@ module.exports = {
         "user": "bucketlist",
         "password": process.env.MYSQL_PASSWORD,
         "database": "bucketlist",
-        "ssl": {
-          "key": fs.readFileSync(`${configFolder}/ssl_cert/client-key.pem`),
-          "cert": fs.readFileSync(`${configFolder}/ssl_cert/client-cert.pem`),
-          "ca": fs.readFileSync(`${configFolder}/ssl_cert/server-ca.pem`)
-        }
```
Comment (karenfrilya97) [inline src/server/config/settings.js]: should use ssl so it's more secure

File: build-utils/webpack.common.js [hunk 7]
```diff
@@ -1,46 +1,58 @@
...
-// and expose it for the front-end via Webpack
-// https://devcenter.heroku.com/articles/dyno-metadata#usage
-envs["APP_NAME"] = process.env.HEROKU_APP_NAME;
-
-// check if http security. if on heroku, enfore https via envs
-envs["IS_HEROKU"] = (process.env.HEROKU_APP_NAME) ? true : false;
+// here we expose the API_URL as env variable; and to be build specific
+envs["API_URL"] = (isHeroku())
+  ? `https://${herokuAppName}.herokuapp.com/api/v1`
+  : `http://localhost:3001/api/v1`;
```
Comment (karenfrilya97) [inline build-utils/webpack.common.js]: good! makes it neater

File: build-utils/webpack.dev.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+const { DefinePlugin } = require('webpack');
+
+module.exports = {
+  mode: 'development',
+  devtool: 'eval-source-map',
+  plugins: [
+    new DefinePlugin({
+      'process.env': {
+        'NODE_ENV': JSON.stringify('development'),
```
Comment (karenfrilya97) [inline build-utils/webpack.dev.js]: this extra comma will cause an error right?

**Pull request closed without merging**